pkg/middleware/validation.go

```diff
@@ -45,31 +48,50 @@ func NewValidationMiddleware(validatorHook string) mux.MiddlewareFunc {
 	}
 }
 
+func maybeLogValidationReason(valid ValidationResponse, r *http.Request, mod string, version string) {
+	if len(valid.Message) > 0 {
+		entry := log.EntryFromContext(r.Context())
+		entry.Warnf("Module %s@%s %s", mod, version, valid.Message)
+	}
+}
+
 type validationParams struct {
 	Module  string
 	Version string
 }
 
-func validate(hook, mod, ver string) (bool, error) {
+type ValidationResponse struct {
+	Status  int
```

Comment:
I think it might better to change this to `Valid bool`

Reply:
https://github.com/gomods/athens/pull/1517/commits/12dba10897b22c735db29b02d50fd64a85bba2ea